pkg/virt-handler/vm.go

```diff
@@ -439,36 +439,57 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 
 	// Cacluate whether the VM is migratable
 	if !condManager.HasCondition(vmi, v1.VirtualMachineInstanceIsMigratable) {
-		isBlockMigration, err := d.checkVolumesForMigration(vmi)
-		liveMigrationCondition := v1.VirtualMachineInstanceCondition{
-			Type:   v1.VirtualMachineInstanceIsMigratable,
-			Status: k8sv1.ConditionTrue,
+		migrate := true
+
+		// Hot fix of https://bugzilla.redhat.com/show_bug.cgi?id=1760028
+		err := d.checkCPUForMigration(vmi)
+		if err != nil {
+			liveMigrationCondition := v1.VirtualMachineInstanceCondition{
+				Type:    v1.VirtualMachineInstanceIsMigratable,
+				Status:  k8sv1.ConditionFalse,
+				Message: err.Error(),
+				Reason:  v1.VirtualMachineInstanceReasonCPUNotMigratable,
+			}
+			vmi.Status.Conditions = append(vmi.Status.Conditions, liveMigrationCondition)
+			migrate = false
 		}
+
+		isBlockMigration, err := d.checkVolumesForMigration(vmi)
 		if err != nil {
-			liveMigrationCondition.Status = k8sv1.ConditionFalse
-			liveMigrationCondition.Message = err.Error()
-			liveMigrationCondition.Reason = v1.VirtualMachineInstanceReasonDisksNotMigratable
+			liveMigrationCondition := v1.VirtualMachineInstanceCondition{
+				Type:    v1.VirtualMachineInstanceIsMigratable,
+				Status:  k8sv1.ConditionFalse,
+				Message: err.Error(),
+				Reason:  v1.VirtualMachineInstanceReasonDisksNotMigratable,
+			}
 			vmi.Status.Conditions = append(vmi.Status.Conditions, liveMigrationCondition)
+			migrate = false
 		}
+
 		err = d.checkNetworkInterfacesForMigration(vmi)
 		if err != nil {
-			liveMigrationCondition = v1.VirtualMachineInstanceCondition{
+			liveMigrationCondition := v1.VirtualMachineInstanceCondition{
 				Type:    v1.VirtualMachineInstanceIsMigratable,
 				Status:  k8sv1.ConditionFalse,
 				Message: err.Error(),
 				Reason:  v1.VirtualMachineInstanceReasonInterfaceNotMigratable,
 			}
 			vmi.Status.Conditions = append(vmi.Status.Conditions, liveMigrationCondition)
-		}
-		if liveMigrationCondition.Status == k8sv1.ConditionTrue {
-			vmi.Status.Conditions = append(vmi.Status.Conditions, liveMigrationCondition)
+			migrate = false
 		}
 
-		// Set VMI Migration Method
-		if isBlockMigration {
-			vmi.Status.MigrationMethod = v1.BlockMigration
-		} else {
-			vmi.Status.MigrationMethod = v1.LiveMigration
+		if migrate {
```

Comment:
Just a personal preference here so feel free to keep it as is, but I'd rather checking `if !migrate` and return here. Then, the entire block below will not be indented.
But again, just a personal preference...

Reply:
@danielBelenky thanks for the review :). I rather would leave the code as is.